### PR TITLE
Enabled non-consuming fallible conversion from GDAL to geo-types Geometry

### LINKED ISSUE #276: Exposing `gdal_to_geo`
Is there any reason why `gdal_to_geo` can't be made public? Either through the `vector` module, or through some other "utils" module?

https://github.com/georust/gdal/blob/fb6e260d46409cb5842c7c2f4978bb5d7bb2ebb0/src/vector/mod.rs#L21

### PR DESCRIPTION
**Pull request by metasim**
Also added `Geometry::to_geo` method, for symmetry with `ToGdal::to_gdal`.

- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Closes #276

### CODE REVIEW

File: src/vector/gdal_to_geo.rs [hunk 1]
```diff
@@ -110,3 +110,10 @@ impl TryFrom<Geometry> for geo_types::Geometry<f64> {
         }
     }
 }
+
+impl TryFrom<Geometry> for geo_types::Geometry<f64> {
+    type Error = GdalError;
+    fn try_from(value: Geometry) -> Result<Self, Self::Error> {
+        TryFrom::<&Geometry>::try_from(&value)
```
Comment (lnicola) [inline src/vector/gdal_to_geo.rs:117]:
```suggestion
        Self::try_from(&value)
```

Comment (lnicola) [reply src/vector/gdal_to_geo.rs:117]: Is my suggestion somehow wrong?

I'm a bit confused, you :+1:ed it, marked it as resolved, but didn't change the code.

Comment (metasim) [reply src/vector/gdal_to_geo.rs]: @lnicola I forgot to `push` 🤦 . I'm recovering from _The Covid_, and not only forgot, but did them in the wrong order.  My apologies.

Comment (lnicola) [reply src/vector/gdal_to_geo.rs]: No worries, and good luck with your recovery!

**Pull request merged**